c7n/resources/appsync.py

```diff
@@ -0,0 +1,170 @@
+              - name: filter-graphql-api-wafv2-regex
+                resource: graphql-api
+                filters:
+                  - type: wafv2-enabled
+                    state: false
+                    web-acl: .*FMManagedWebACLV2-?FMS-.*
+    """
+
+    schema = type_schema(
+        'wafv2-enabled', **{
+            'web-acl': {'type': 'string'},
+            'state': {'type': 'boolean'}})
+
+    permissions = ('wafv2:ListWebACLs',)
+
+    def process(self, resources, event=None):
+        wafs = self.manager.get_resource_manager('wafv2').resources(augment=False)
+        waf_name_id_map = {w['Name']: w['ARN'] for w in wafs}
+
+        target_acl = self.data.get('web-acl', '')
+        state = self.data.get('state', False)
+        target_acl_ids = [v for k, v in waf_name_id_map.items() if
+                          re.match(target_acl, k)]
+
+        results = []
+        for r in resources:
+            r_web_acl_id = r.get('wafWebAclArn')
+            if state:
+                if not target_acl and r_web_acl_id:
+                    results.append(r)
+                elif target_acl and r_web_acl_id in target_acl_ids:
+                    results.append(r)
+            else:
+                if not target_acl and not r_web_acl_id:
+                    results.append(r)
+                elif target_acl and r_web_acl_id not in target_acl_ids:
+                    results.append(r)
+        return results
+
+
+@GraphQLApi.action_registry.register('set-wafv2')
+class SetWafv2(BaseAction):
+    """Enable wafv2 protection on AppSync graphqlApi.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-wafv2-for-graphql-api
+                resource: graphql-api
+                filters:
+                  - type: wafv2-enabled
+                    state: false
+                    web-acl: test-waf-v2
+                actions:
+                  - type: set-wafv2
+                    state: true
+                    force: true
+                    web-acl: test-waf-v2
+
+              - name: unset-wafv2-for-graphql-api
+                resource: graphql-api
+                filters:
+                  - type: wafv2-enabled
+                    state: true
+                actions:
+                  - type: set-wafv2
+                    state: true
+                    force: true
+                    web-acl: test-waf-v2
+
+            policies:
+              - name: set-wafv2-for-graphql-api-regex
+                resource: graphql-api
+                filters:
+                  - type: wafv2-enabled
+                    state: false
+                    web-acl: .*FMManagedWebACLV2-?FMS-.*
+                actions:
+                  - type: set-wafv2
+                    state: true
+                    force: true
+                    web-acl: FMManagedWebACLV2-?FMS-TestWebACL
+    """
+    permissions = ('appsync:UpdateGraphqlApi', 'wafv2:ListWebACLs')
```

Comment:
missing permissions on wafv2 for associate/dissasociate afaics

Reply:
@kapilt oops, permissions updated.